extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -411,147 +539,199 @@ public Integer get() {
-                                            (int) Math.min(keyCount.intValue() * 2L, maxKeys)));
-                                }
-                                keys.getValue()[keyCount.intValue()] = toParquetPrimitive.apply(o);
-                                Integer result = keyCount.getValue();
-                                keyCount.increment();
-                                return result;
-                            }));
+        }
+    }
 
-                        }
-                        buffersPerPage.add(posInDictionary);
-                    }
-                }
-                List<IntBuffer> repeatCount = null;
-                if (lengthSource != null) {
-                    repeatCount = new ArrayList<>();
-                    try (final ChunkSource.GetContext context = lengthSource.makeGetContext(targetSize);
-                            final RowSequence.Iterator it = lengthRowSet.getRowSequenceIterator()) {
-                        while (it.hasMore()) {
-                            final RowSequence rs = it.getNextRowSequenceWithLength(rowStepGetter.get());
-                            // noinspection unchecked
-                            final IntChunk<Values> chunk = (IntChunk<Values>) lengthSource.getChunk(context, rs);
-                            final IntBuffer newBuffer = IntBuffer.allocate(chunk.size());
-                            chunk.copyToTypedBuffer(0, newBuffer, 0, chunk.size());
-                            newBuffer.limit(chunk.size());
-                            repeatCount.add(newBuffer);
-                        }
-                    }
-                }
-                columnWriter.addDictionaryPage(keys.getValue(), keyCount.intValue());
-                final Iterator<IntBuffer> repeatCountIt = repeatCount == null ? null : repeatCount.iterator();
-                for (final IntBuffer intBuffer : buffersPerPage) {
-                    intBuffer.flip();
-                    if (lengthSource != null) {
-                        columnWriter.addVectorPage(intBuffer, repeatCountIt.next(), intBuffer.remaining(),
-                                Integer.MIN_VALUE);
-                    } else if (hasNulls.getValue()) {
-                        columnWriter.addPage(intBuffer, Integer.MIN_VALUE, intBuffer.remaining());
+    private static <DATA_TYPE> void encodePlain(Map<String, Map<CacheTags, Object>> computedCache,
+            TrackingRowSet tableRowSet,
+            ColumnDefinition<DATA_TYPE> columnDefinition,
+            ParquetInstructions writeInstructions,
+            RowSet rowSet,
+            ColumnSource<DATA_TYPE> columnSource,
+            ColumnSource<?> lengthSource,
+            RowSet lengthRowSet,
+            LongSupplier rowStepGetter,
+            LongSupplier valuesStepGetter,
+            int targetRowsPerPage,
+            int pageCount,
+            Class<DATA_TYPE> columnType,
+            ColumnWriter columnWriter) throws IOException {
```

Comment:
Since we're adding `final`s and `@NotNull`s, maybe the two new methods should get the same treatment?